File: net/src/Sdl.Web.GraphQL/Request/GraphQLRequest.cs
using System.Collections.Generic;
using System.Runtime.Serialization;
using Sdl.Web.HttpClient.Auth;
using Newtonsoft.Json;
using Sdl.Web.HttpClient;

namespace Sdl.Web.GraphQLClient.Request
{
    public class GraphQLRequest : IGraphQLRequest
    {
        [JsonIgnore]
        public IAuthentication Authenticaton { get; set; }
        public string OperationName { get; set; }
        public string Query { get; set; }
        public IDictionary<string, object> Variables { get; set; } = new Dictionary<string, object>();
        public HttpHeaders Headers { get; set; }
        public SerializationBinder Binder { get; set; }
        public List<JsonConverter> Convertors { get; set; } = new List<JsonConverter>();
        public IGraphQLRequest AddVariable(string name, object value)
        {
            Variables.Add(name, value);
            return this;
        }
    }
}

File: java/GraphLClient/src/main/java/com/sdl/web/Request/GraphQLRequest.java
package com.sdl.web.Request;

import com.sdl.web.Auth.IAuthentication;

import java.util.Dictionary;

public class GraphQLRequest implements IGraphQLRequest{
    public IAuthentication Authenticaton;
    public String OperationName;
    public String Query;
    public Dictionary<String, Object> Variables;
    /*public SerializationBinder Binder;*/
   /* public List<JsonConverter> Convertors;*/

    @Override
    public void setAuthenticaton(IAuthentication authenticaton) {
        Authenticaton = authenticaton;
    }
    @Override
    public IAuthentication getAuthenticaton() {
        return Authenticaton;
    }
    @Override
    public void setOperationName(String operationName) {
        OperationName = operationName;
    }
    @Override
    public String getOperationName() {
        return OperationName;
    }
    @Override
    public void setQuery(String query) {
        Query = query;
    }
    @Override
    public String getQuery() {
        return Query;
    }
    @Override
    public void setVariables(Dictionary<String, Object> variables) {
        Variables = variables;
    }
    @Override
    public Dictionary<String, Object> getVariables() {
        return Variables;
    }

    public IGraphQLRequest AddVariable(String name, Object value)
    {
        Variables.put(name, value);
        return this;
    }
}
